README.MD

```diff
@@ -40,7 +40,22 @@ npm config set https-proxy http://<username>:<password>@<the.proxy.hostname>:<po
 npm config set no-proxy localhost,127.0.0.1,example.org
 ```
 
-Additionally proxy settings found in the environment variables `HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY` will be used if they are not defined in the `.npmrc` file. 
+Additionally proxy settings found in the environment variables `HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY` will be used if they are not defined in the `.npmrc` file.
+
+### Cache Downloaded Binaries
+By default downloaded chromium binaries are not cached on the filesystem. You may enable caching by setting the `CHROMIUM_CACHE` environment variable to a directory path, for example:
+
+```bash
+export CHROMIUM_CACHE=/path/to/cache/rootdir/
```

Comment:
I'm thinking about the usability for users. Might it be better if we provide a possibility to turn on caching without the need of specifying path?
E.g. having 2 variables:
```
export CHROMIUM_CACHE_ENABLED
export CHROMIUM_CACHE
```
So, when user wants to turn on caching but doesn't want to bother with paths. We can use something default like `.cache` folder where we will save all archives we download. It's even possible to unzip from there instead of tmp path as we do know.

It's more of a usability question and your opinion. For me, it's always much harder to set folder path than to enable a feature. When I'm asked to set a path, I should know if this is absolute / relative path, access is allowed / denied, should I pre-create folder or it will be created by tool, should this folder contain some specific structure, e.t.c.

I'm not fully sure about this feature neither so it would be great if you tell what you think about this.

From what I can see, there might be next use cases for this feature:
1. As described in original request, person wants to have cache at CI system / reuse `node_chromium` folder (though I don't understand how is this possible as most dependencies are actually OS/arch dependant).
2. Testing our own module (much easier to perform checks without the need of installation)
3. When update of module is done (I assume in this case `postinstall` is called as well) - this one looks the most often used and the most important for me

According to **3.** this feature might be very useful if we enable it by default. What do you think about this?

Reply:
1. I could see folder reuse happening using docker or perhaps network shares. I replicated it like this:

```bash
docker run -it -v /Users/rick/projects:/projects node /bin/bash
cd /projects/node-chromium
npm run test
````

Then exit docker and run:

```bash
npm install

ls lib/chromium/
# outputs: chrome-linux/ chrome-mac/
```

There may be other scenarios using networked drives or VMs...

2. Yes this makes running the tests much faster and we could potentially avoid the download completely, though that would probably involve putting a chromium archive under source control.

3. YES! You are totally right, this should be on by default. I will rework it to be so.